### ISSUE TITLE
Random function

### ISSUE THREAD

**Issue by vanillajonathan** [PRQL/prql#4661]
### What's up?

PRQL does not yet have any random function(s). SQL have the `RAND` function which can be used awkwardly used using s-strings.

It would be nice if PRQL has a random function, it could work like [`randint`](https://docs.python.org/3/library/random.html#random.randint) in Python, i.e. take in a low and high parameter in order to specify a range. The SQL `RAND` function does not have this feature it can takes a seed as parameter.

```elm
derive { random_value = (random low:100 high:1000) }
```

```sql
SELECT FLOOR(100 + RAND() * (1000 - 100 + 1)) AS random_value;
```

It could even generate a random letter:
```elm
derive { random_letter = (random low:A high:Z) }
```

```sql
SELECT CHAR(FLOOR(RAND() * (ASCII('A') - ASCII('Z')))) AS random_letter;
```

**Comment by max-sixty** [comment]
Yes, would be happy to add this. Probably start with `math.randint`?

@vanillajonathan if you want to add, search for `math.abs` as an example.

**Comment by aljazerzen** [comment]
Note that random is an impure function and suffers from the problem described in #1111
